feature_engine/wrappers/wrappers.py

```diff
@@ -357,15 +369,18 @@ def get_feature_names_out(self) -> List:
         if self.transformer_.__class__.__name__ in _TRANSFORMERS:
             feature_names = self.feature_names_in_
 
-        if self.transformer_.__class__.__name__ in _CREATORS:
+        elif self.transformer_.__class__.__name__ in _CREATORS:
             added_features = self.transformer_.get_feature_names_out(self.variables_)
             feature_names = list(self.feature_names_in_) + list(added_features)
 
-        if self.transformer_.__class__.__name__ in _SELECTORS:
+        elif self.transformer_.__class__.__name__ in _SELECTORS:
             feature_names = [
                 f for f in self.feature_names_in_ if f not in self.features_to_drop_
             ]
 
+        else:
```

Comment:
we don't need this else, we need to modify the ifs and elifs behaviours.

If input_features is None, the behaviour is the current one, but if input_features is a list, we need to check that all the elements of that list are within self.variables_ and if yes, then return added_features passing input_features instead of self.variables_

Reply:
I tried to implement your suggestion. Hopefully, it is almost correct.